src/V3Randomize.cpp

```diff
@@ -438,13 +456,97 @@ class ConstraintExprVisitor final : public VNVisitor {
     }
 };
 
-template <typename TreeNodeType>
-class CaptureFrame final {
-    TreeNodeType* m_treep;  // Original tree
+class ClassLookupHelper final {
+    const std::set<AstNodeModule*> m_visibleModules;
+    const std::vector<AstNodeModule*> m_symLookupOrder;
+    std::map<AstNode*, AstNodeModule*> m_classMap;
+
+    using inits_t = std::tuple<std::set<AstNodeModule*>, std::vector<AstNodeModule*>>;
+
+    // BFS search
+    template <typename Action>
+    static void foreachSuperClass(AstClass* classp, Action action) {
+        std::queue<AstClass*> classes;
+        classes.push(classp);
+        while (!classes.empty()) {
+            classp = classes.front();
+            classes.pop();
+            for (AstClassExtends* extendsp = classp->extendsp(); extendsp;
+                 extendsp = VN_AS(extendsp->nextp(), ClassExtends)) {
+                AstClass* const superClassp
+                    = VN_AS(extendsp->childDTypep(), ClassRefDType)->classp();
+                action(superClassp);
+                classes.push(superClassp);
+            }
+        }
+    }
+
+    static inits_t init(AstClass* classp) {
+        std::set<AstNodeModule*> visibleModules = {classp};
+        std::vector<AstNodeModule*> symLookupOrder = {classp};
+        foreachSuperClass(classp, [&](AstClass* superclassp) {
+            visibleModules.emplace(superclassp);
+            symLookupOrder.push_back(superclassp);
+        });
+        return inits_t(visibleModules, symLookupOrder);
+    }
+
+    ClassLookupHelper(inits_t inits)
+        : m_visibleModules(std::get<0>(inits))
+        , m_symLookupOrder(std::get<1>(inits)) {}
+
+public:
+    bool moduleInClassHierarchy(AstNodeModule* modp) const { return m_visibleModules.count(modp); }
```

Comment:
```suggestion
    bool moduleInClassHierarchy(AstNodeModule* modp) const { return m_visibleModules.count(modp) != 0; }
```

To prevent lint warning of loss of precision

Reply:
Done. I'm not getting such lints, but I'm not using the provided `.clang-tidy` as it causes a lot of bizarre unwanted lints.